From: Muchun Song <songmuchun@bytedance.com>
Subject: arm64: mm: hugetlb: restructure the code comment inside flush_dcache_page()

Restructure the code comment inside flush_dcache_page() to make it more
clear.

Link: https://lkml.kernel.org/r/20220414072646.21910-1-songmuchun@bytedance.com
Signed-off-by: Muchun Song <songmuchun@bytedance.com>
Reviewed-by: Anshuman Khandual <anshuman.khandual@arm.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/arm64/mm/flush.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

--- a/arch/arm64/mm/flush.c~arm64-mm-hugetlb-enable-hugetlb_page_free_vmemmap-for-arm64-fix
+++ a/arch/arm64/mm/flush.c
@@ -81,9 +81,10 @@ void flush_dcache_page(struct page *page
 	 * read-only when CONFIG_HUGETLB_PAGE_FREE_VMEMMAP is enabled (more
 	 * details can refer to vmemmap_remap_pte()).  Although
 	 * __sync_icache_dcache() only set PG_dcache_clean flag on the head
-	 * page struct, some tail page structs still can be seen the flag is
-	 * set since the head vmemmap page frame is reused (more details can
-	 * refer to the comments above page_fixed_fake_head()).
+	 * page struct, there is more than one page struct with PG_dcache_clean
+	 * associated with the HugeTLB page since the head vmemmap page frame
+	 * is reused (more details can refer to the comments above
+	 * page_fixed_fake_head()).
 	 */
 	if (hugetlb_free_vmemmap_enabled() && PageHuge(page))
 		page = compound_head(page);
_
